oojs/application.js

```diff
@@ -1,13 +1,44 @@
+var dice = [];
+var dieNumber = 0;
+
 $(document).ready(function() {
+  var diceSet = new Dice();
+
   $('#roller button.add').on('click', function() {
-    console.log("WAT")
-    $('.dice').append('<div class="die">0</div>');
+    diceSet.add();
   });
 
   $('#roller button.roll').on('click', function() {
-    $('.die').each(function(k, die) {
-      var value = Math.floor((Math.random()*6)+1);
-      $(die).text(value);
-    });
+    diceSet.rollAll();
   });
 });
+
+
+// Dice group object
+function Dice() {
+  this.dice = []
+
+  this.add = function() {
+    this.dice.push(new Die());
+  };
+
+  this.rollAll = function() {
+    var index;
+    for (index = 0; index < this.dice.length; ++index) {
+      this.dice[index].roll();
+    }
+  };
+};
+
+// Die object
+function Die() {
+  this.value = 0;
+  this.id = "die_" + dieNumber.toString();
+  dieNumber++;
+  $('.dice').append('<div class="die" id="' + this.id + '">' + this.value + '</div>');
```

Comment:
Save this as an attribute so you don't need an `id` attribute. You can just modify the attribute and it updates the HTML automatically for you.